functions/httpserver.go

```diff
@@ -211,5 +216,96 @@ func nodePeers(c *gin.Context) {
 		return
 	}
 	c.JSON(http.StatusOK, peers)
+}
+
+func join(c *gin.Context) {
+	joinReq := RegisterSSO{}
+	if err := c.BindJSON(&joinReq); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "could not parse request" + err.Error()})
+		return
+	}
+	if err := RegisterWithSSO(&joinReq); err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err})
+		return
+	}
+	c.JSON(http.StatusOK, nil)
+}
 
+func sso(c *gin.Context) {
+	registerData := RegisterSSO{}
+	if err := c.BindJSON(&registerData); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "could not parse request" + err.Error()})
+		return
+	}
+	socketUrl := fmt.Sprintf("wss://%s/api/v1/auth-register/host", registerData.API)
+	// Dial the netmaker server controller
+	conn, _, err := websocket.DefaultDialer.Dial(socketUrl, nil)
+	if err != nil {
+		logger.Log(0, fmt.Sprintf("error connecting to %s : %s", registerData.API, err.Error()))
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err})
+		return
+	}
+	host := hostForSSO()
+	request := models.RegisterMsg{
+		RegisterHost: host,
+		User:         registerData.User,
+		Password:     registerData.Pass,
+		Network:      registerData.Network,
+		JoinAll:      registerData.AllNetworks,
+	}
+	registerData.Pass = ""
+	defer conn.Close()
+	reqData, err := json.Marshal(&request)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err})
+		return
+	}
+	if err := conn.WriteMessage(websocket.TextMessage, reqData); err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err})
+		return
+	}
+	for {
+		msgType, msg, err := conn.ReadMessage()
+		if err != nil {
+			if msgType < 0 {
+				logger.Log(1, "received close message from server")
+				c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+				return
+			}
+			if !strings.Contains(err.Error(), "normal") { // Error reading a message from the server
+				logger.Log(0, "read:", err.Error())
+				c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+				return
+			}
+		}
+		if msgType == websocket.CloseMessage {
+			logger.Log(1, "received close message from server")
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "websocket closed"})
+			return
+		}
+		if strings.Contains(string(msg), "oauth/register") { // TODO: maybe send to channel for GUI in future?
+			c.JSON(http.StatusOK, gin.H{"authendpoint": string(msg)})
+			return
+		}
+	}
+}
+
+func hostForSSO() models.Host {
+	host := config.Netclient()
+	ip, err := getInterfaces()
+	if err != nil {
```

Comment:
why is `getInterfaces()` called here, how is this related to SSO?

Reply:
following the pattern from logic.Register()